From patchwork Fri Apr  8 17:00:07 2016
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [U-Boot] arc: Add virt_to_phys() stub
From: Alexey Brodkin <Alexey.Brodkin@synopsys.com>
X-Patchwork-Id: 608123
Message-Id: <1460134807-16503-1-git-send-email-abrodkin@synopsys.com>
To: u-boot@lists.denx.de
Cc: Marek Vasut <marex@denx.de>, Alexey Brodkin <Alexey.Brodkin@synopsys.com>
Date: Fri,  8 Apr 2016 10:00:07 -0700

Commit cf7c93cdd755 "usb: ehci: Implement V2P mapping"
introduced usage of virt_to_phys() in ehci-hcd.

Since there was no implementation of virt_to_phys() for ARC
compilation of the ehci-generic driver failed.

This change adds virt_to_phys() stub for ARC so now
USB driver for AXS101 board could be built again.

Signed-off-by: Alexey Brodkin <abrodkin@synopsys.com>
Cc: Marek Vasut <marex@denx.de>
Cc: Daniel Schwierzeck <daniel.schwierzeck@gmail.com>
Cc: Hans de Goede <hdegoede@redhat.com>
Acked-by: Marek Vasut <marex@denx.de>
---
 arch/arc/include/asm/io.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arc/include/asm/io.h b/arch/arc/include/asm/io.h
index 281682c..b6f7724 100644
--- a/arch/arc/include/asm/io.h
+++ b/arch/arc/include/asm/io.h
@@ -239,4 +239,9 @@ static inline int __raw_writesl(unsigned int addr, void *data, int longlen)
 #define setbits_8(addr, set) setbits(8, addr, set)
 #define clrsetbits_8(addr, clear, set) clrsetbits(8, addr, clear, set)
 
+static inline phys_addr_t virt_to_phys(void *vaddr)
+{
+	return (phys_addr_t)((unsigned long)vaddr);
+}
+
 #endif	/* __ASM_ARC_IO_H */
